pkg/lint/load.go

```diff
@@ -199,7 +199,14 @@ func (cl *ContextLoader) loadPackages(ctx context.Context, loadMode packages.Loa
 	cl.debugf("Built loader args are %s", args)
 	pkgs, err := packages.Load(conf, args...)
 	if err != nil {
-		return nil, errors.Wrap(err, "failed to load program with go/packages")
+		return nil, errors.Wrap(err, "failed to load with go/packages")
+	}
+
+	// Currently, go/packages doesn't guarantee that error will be returned
+	// if context was cancelled. See
+	// https://github.com/golang/tools/commit/c5cec6710e927457c3c29d6c156415e8539a5111#r39261855
+	if ctx.Err() != nil {
```

Comment:
Perhaps:
```golang
if err := ctx.Err(); err != nil {
```
would do? It would allow to avoid calling `ctx.Err()` twice.

Reply:
I don't see any profit to save nanoseconds here :)

Reply:
upate: maybe microseconds

```
func (c *cancelCtx) Err() error {
	c.mu.Lock()
	err := c.err
	c.mu.Unlock()
	return err
}
```

It doesn't make sense to waste time on such micro-optimization, there are a lot of things to save seconds of time